agent/discovery_chain_endpoint.go

```diff
@@ -0,0 +1,70 @@
+package agent
+
+import (
+	"fmt"
+	"net/http"
+	"strings"
+	"time"
+
+	"github.com/hashicorp/consul/agent/structs"
+)
+
+func (s *HTTPServer) ConnectDiscoveryChainGet(resp http.ResponseWriter, req *http.Request) (interface{}, error) {
+	var args structs.DiscoveryChainRequest
+	if done := s.parse(resp, req, &args.Datacenter, &args.QueryOptions); done {
+		return nil, nil
+	}
+
+	args.Name = strings.TrimPrefix(req.URL.Path, "/v1/connect/discovery-chain/")
+	if args.Name == "" {
+		resp.WriteHeader(http.StatusBadRequest)
+		fmt.Fprint(resp, "Missing chain name")
+		return nil, nil
+	}
+
+	args.EvaluateInDatacenter = req.URL.Query().Get("eval_dc")
+	// TODO(namespaces): args.EvaluateInNamespace = req.URL.Query().Get("eval_namespace")
+
+	overrideMeshGatewayMode := req.URL.Query().Get("override_mesh_gateway_mode")
+	if overrideMeshGatewayMode != "" {
+		mode, err := structs.ValidateMeshGatewayMode(overrideMeshGatewayMode)
+		if err != nil {
+			resp.WriteHeader(http.StatusBadRequest)
+			fmt.Fprint(resp, "Invalid override_mesh_gateway_mode parameter")
+			return nil, nil
+		}
+		args.OverrideMeshGateway.Mode = mode
+	}
+
+	args.OverrideProtocol = req.URL.Query().Get("override_protocol")
+	overrideTimeoutString := req.URL.Query().Get("override_connect_timeout")
```

Comment:
This seems like a strange addition to me. If the client wants to override one field can't they just do that in the resposne they get back? It's not an input to the RPC. Ditto for mesh gateway mode above.

Maybe I'm missing some important use-case though? I'd lean towards leaving it out unless it's vital as it won't scale well as we add more config options in here to add more override and transforms into the endpoint.